pinot-common/src/main/java/org/apache/pinot/common/response/BrokerResponse.java

```diff
@@ -32,264 +34,241 @@
 public interface BrokerResponse {
 
   /**
-   * Set exceptions caught during request handling, into the broker response.
+   * Convert the broker response to JSON String.
    */
-  void setExceptions(List<ProcessingException> exceptions);
-
-  void addToExceptions(QueryProcessingException processingException);
+  default String toJsonString()
+      throws IOException {
+    return JsonUtils.objectToString(this);
+  }
 
   /**
-   * Set the number of servers got queried by the broker.
-   *
-   * @param numServersQueried number of servers got queried.
+   * Returns the result table.
    */
-  void setNumServersQueried(int numServersQueried);
+  @Nullable
+  ResultTable getResultTable();
 
   /**
-   * Set the number of servers responded to the broker.
-   *
-   * @param numServersResponded number of servers responded.
+   * Sets the result table. We expose this method to allow modifying the results on the client side, e.g. hiding the
+   * results and only showing the stats.
    */
-  void setNumServersResponded(int numServersResponded);
+  void setResultTable(@Nullable ResultTable resultTable);
```

Comment:
This seems to be the only mutable method in this interface. Wouldn't be better to implement the case shown in the example using a delegate/decorate pattern? Basically having an implementation that delegates all methods but `getResultTable` and `getNumRowsResultSet` to a another `BrokerResponse` stored as attribute

Reply:
I feel that is a little bit overkilling if the only benefit is getting rid of this setter. I'm not a big fan of this only setter as well, but I feel it is much simpler and easier to understand if we want to allow client to override it